heliopy/data/mms.py

```diff
@@ -7,131 +7,1773 @@
+    query['TAIENDTIME<'] = '{0:d}'.format(t1)
+
+    # Post the query
+    sesh = requests.Session()
+    sesh.auth = (sdc_username, sdc_password)
+    data = sesh.get(url_path, params=query)
+
+    # Read first line as dict keys. Cut text from TAI keys
+    data = _response_text_to_dict(data.text)
+
+    # Convert to useful types
+    types = ['int16', 'int64', 'int64', 'str', 'float32', 'int8',
+             'int8', 'str', 'int32', 'str', 'datetime', 'datetime',
+             'int32', 'int32', 'int32', 'int32', 'int32', 'int32',
+             'int32', 'int32', 'int32', 'int32', 'int32', 'str']
+    for items in zip(data, types):
+        if items[1] == 'str':
+            pass
+        elif items[1] == 'datetime':
+            data[items[0]] = [dt.datetime.strptime(value,
+                                                   '%Y-%m-%d %H:%M:%S'
+                                                   )
+                              if value != '' else value
+                              for value in data[items[0]]
+                              ]
+        else:
+            data[items[0]] = np.asarray(data[items[0]], dtype=items[1])
+
+    # Add useful tags
+    #   - Number of seconds elapsed
+    #   - TAISTARTIME as datetime
+    #   - TAIENDTIME as datetime
+    data['TAISTARTTIME'] = data.pop('TAISTARTTIME '
+                                    '(TAI seconds since 1958-01-01)'
+                                    )
+    data['TAIENDTIME'] = data.pop('TAIENDTIME '
+                                  '(TAI seconds since 1958-01-01)'
+                                  )
+    data['DT'] = data['TAIENDTIME'] - data['TAISTARTTIME']
+
+    # NOTE! If data['TAISTARTTIME'] is a scalar, this will not work
+    #       unless everything after "in" is turned into a list
+    data['TSTART'] = [dt.datetime(
+                         *value[0:6], value[6]*1000+value[7]
+                         )
+                      for value in
+                      epochs.CDFepoch.breakdown_tt2000(
+                          data['TAISTARTTIME']*int(1e9)+t_1958
+                          )
+                      ]
+    data['TEND'] = [dt.datetime(
+                         *value[0:6], value[6]*1000+value[7]
+                         )
+                    for value in
+                    epochs.CDFepoch.breakdown_tt2000(
+                        data['TAIENDTIME']*int(1e9)+t_1958
+                        )
+                    ]
+
+    return data
+
+
+def construct_file_names(*args, data_type='science', **kwargs):
+    '''
+    Construct a file name compliant with MMS file name format guidelines.
+
+    MMS file names follow the convention
+        sc_instr_mode_level[_optdesc]_tstart_vX.Y.Z.cdf
+
+    Parameters
+    ----------
+        args : dict
+            Arguments to be passed along.
+        data_type : str
+            Type of file names to construct. Options are:
+            science or *_selections. If science, inputs are
```

Comment:
If you want a *, you have to escape it (ie. do `\*`, since `*` is used by rst as the emphasis marker.